File: app-v3/src/test/java/com/khmelenko/lab/varis/TestDateTimeUtils.java
package com.khmelenko.lab.varis;

import com.khmelenko.lab.varis.util.DateTimeUtils;

import org.junit.Test;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;
import java.util.Locale;

import static junit.framework.TestCase.assertEquals;

/**
 * Testing DateTimeUtils class
 *
 * @author Dmytro Khmelenko (d.khmelenko@gmail.com)
 */
public class TestDateTimeUtils {

    @Test
    public void testParseXmlDateTime() throws ParseException {
        Calendar expected = new GregorianCalendar();
        DateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ", Locale.getDefault());
        String xmlDate = formatter.format(expected.getTime());
        Date actual = DateTimeUtils.parseXmlDateTime(xmlDate);

        assertEquals(expected.getTime().getTime() / 1000, actual.getTime() / 1000);
    }

}

File: app-v3/src/test/java/com/khmelenko/lab/varis/TestDateTimeUtils.kt
package com.khmelenko.lab.varis

import com.khmelenko.lab.varis.util.DateTimeUtils
import junit.framework.TestCase.assertEquals
import org.junit.Test
import java.text.SimpleDateFormat
import java.util.GregorianCalendar
import java.util.Locale

/**
 * Testing DateTimeUtils class
 *
 * @author Dmytro Khmelenko (d.khmelenko@gmail.com)
 */
class TestDateTimeUtils {

    @Test
    fun testParseXmlDateTime() {
        val expected = GregorianCalendar()
        val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ", Locale.getDefault())
        val xmlDate = formatter.format(expected.time)
        val actual = DateTimeUtils.parseXmlDateTime(xmlDate)

        assertEquals(expected.time.time / 1000, actual.time / 1000)
    }
}
